### DISCUSSION TITLE
SafeOnTheBeach (14/01/2022)

### DISCUSSION THREAD

**Discussion by progettistamatto** [sapienzastudentsnetwork/basi-di-dati-2#45] [Esami]
In questa discussion a tema il progetto "SafeOnTheBeach", oggetto dei testi di esame del 14 gennaio 2022 e del 10 ottobre 2024, discuteremo di eventuali proposte di soluzione, dubbi o scelte progettuali in merito alla realizzazione del progetto in questione

**N.B.:** Si ha anche un'altra versione, omonima ma più vecchia, delle specifiche del progetto, oggetto del testo di esame del 4 settembre 2020. Le due versioni differiscono per gli ultimi due use case (n. 7 e n. 8):

  - **Versione più recente (del 14 gennaio 2022 e del 10 ottobre 2024):**
   
     - > Data una posizione (latitudine e longitudine) un istante e un intero positivo $k$, deve essere possibile calcolare le $k$ zone balneari più vicine alla posizione data, ordinate per _densità turistica attesa_, nell'istante dato, crescente. La densità turistica di una zona in un istante (futuro) $t$ è il rapporto tra la capienza della zona e il numero di utenti che si prevede (in base alle prenotazioni già effettuate) saranno presenti nella zona all'istante $i$. (In fase di analisi, si assuma di avere a disposizione un simbolo di funzione distanza/4, soggetto alla semantica di mondo reale, che, date le coordinate geografiche di due punti ($lat_1$, $lon_1$) e ($lat_2$, $lon_2$), valuti alla distanza tra loro espressa in metri. In fase di progettazione, si assuma che sia disponibile una funzione definita nel DBMS che la implementi.)

     - >Data una località $l$ e un periodo di tempo, il sistema deve permettere di calcolare tutte le spiagge che si trovano in $l$ presso le cui attività non è stato organizzato nessun evento all'interno del periodo dato.
  
  -  **Versione meno recente (del 4 settembre 2020):** 
  
     - > Ricerca degli eventi non ancora terminati più gettonati nella zona, cioè con un numero di prenotazioni (una prenotazione conta per il numero di partecipanti del gruppo) più alto della media
     
     - > Data una zona balneare, il sistema deve poter calcolare la disponibilità attuale di posti in tale zona rispetto alla sua capienza massima. Ad esempio, se una zona ha una capienza di $100$ persone e $10$ persone al suo intero, il numero di posti disponibili è $100 - 10 = 90$

**Comment by progettistamatto** [comment]
Per questo esame è disponibile [[QUI]](https://github.com/Exyss/university-notes/raw/main/Bachelor/Secondo%20Anno/Basi%20di%20Dati%20II/SafeOnTheBeach.pdf) una soluzione proposta da Simone Bianco (Exyss)

_**N.B.:** la soluzione in questione riguarda la versione più recente della consegna (14 gennaio 2022) ☑️_

> [!Warning]
_Le soluzioni di Simone Bianco fanno uso degli **Entity-Relationship (ER)** Diagrams su cui il corso si basava fino all'Anno Accademico 2023/2024, mentre **attualmente vengono utilizzati gli UML Class** Diagram. Sebbene i due siano estremamente simili, vi sono **alcune discordanze tra di loro**, in particolare l'**inversione di posizione dei vincoli di molteplicità agli estremi delle relationship**. Nonostante ciò, il rimanente delle fasi di analisi e progettazione delle soluzioni proposte risulta inaffetto da tale cambiamento. Per tanto, una volta comprese le differenze tra ER e Classi UML, le soluzioni risultano normalmente fruibili._

Rispondi a questo commento taggando @Exyss per eventuali dubbi o segnalazioni in merito

**Comment by lele25811** [comment]
Questa è la mia soluzione di SafeOnTheBeach, traccia d'esame del 10 Ottobre 2024, fatemi sapere se ci sono errori.
(Scusate per la qualità del pdf)

[SoluzioneSafeOnTheBeach.pdf](https://github.com/user-attachments/files/18165292/SoluzioneSafeOnTheBeach.pdf)

**Comment by CosaFaSelect1** [comment]
Per questo esame è disponibile [[QUI]](https://github.com/CasuFrost/university-notes-Informatica/raw/main/Secondo%20Anno/Secondo%20Semestre/Basi%20di%20Dati%202/altra%20roba/SafeOnTheBeach.pdf) una soluzione proposta da Marco Casu
